openvdb/points/PointMask.h

```diff
@@ -0,0 +1,384 @@
+    using LeafManagerT = typename tree::LeafManager<TreeT>;
+    using ValueT = typename TreeT::LeafNodeType::ValueType;
+
+    PointsToScalarOp(const PointDataGridT& grid,
+                  const FilterT& filter)
+        : mPointDataAccessor(grid.getConstAccessor())
+        , mFilter(filter) { }
+
+    void operator()(const typename LeafManagerT::LeafRange& range) const {
+
+        for (auto leaf = range.begin(); leaf; ++leaf) {
+
+            const auto* const pointLeaf =
+                mPointDataAccessor.probeConstLeaf(leaf->origin());
+
+            // assumes matching topology
+            assert(pointLeaf);
+
+            for (auto value = leaf->beginValueOn(); value; ++value) {
+                const Index64 count = points::iterCount(
+                    pointLeaf->beginIndexVoxel(value.getCoord(), mFilter));
+                if (count > Index64(0)) {
+                    value.setValue(static_cast<ValueT>(count));
+                }
+                else {
+                    // disable any empty voxels
+                    value.setValueOn(false);
+                }
+            }
+        }
+    }
+
+private:
+    const typename PointDataGridT::ConstAccessor mPointDataAccessor;
+    const FilterT& mFilter;
+}; // struct PointsToScalarOp
+
+
+/// @brief Compute scalar grid from PointDataGrid using a different transform
+///        and while evaluating the point filter
+template <typename GridT, typename PointDataGridT, typename FilterT>
+struct PointsToTransformedScalarOp
+{
+    using LeafManagerT = tree::LeafManager<const typename PointDataGridT::TreeType>;
+    using ValueType = typename GridT::TreeType::LeafNodeType::ValueType;
+    using HandleT = AttributeHandle<Vec3f>;
+    using CombinableT = typename GridCombinerOp<GridT>::CombinableT;
+
+    PointsToTransformedScalarOp(const math::Transform& newTransform,
+                         const math::Transform& transform,
+                         const FilterT& filter,
+                         CombinableT& combinable)
+        : mNewTransform(newTransform)
+        , mTransform(transform)
+        , mFilter(filter)
+        , mCombinable(combinable) {}
+
+    void operator()(const typename LeafManagerT::LeafRange& range) const
+    {
+        bool exists;
+        auto& grid = mCombinable.local(exists);
+        if (!exists) {
+            grid.setTransform(mNewTransform.copy());
+        }
+
+        auto& countTree = grid.tree();
+
+        for (auto leaf = range.begin(); leaf; ++leaf) {
+
+            auto handle = HandleT::create(leaf->template constAttributeArray("P"));
+
+            for (auto iter = leaf->template beginIndexOn(mFilter); iter; iter++) {
+                const Vec3d position = mTransform.indexToWorld(handle->get(*iter) +
+                    iter.getCoord().asVec3d());
+                const Coord ijk = mNewTransform.worldToIndexCellCentered(position);
+
+                auto* countLeaf = countTree.touchLeaf(ijk);
```

Comment:
I would build a local accessor instead of calling touchLeaf:

```
auto& countTree = grid.tree();
tree::ValueAccessor<typename GridT::TreeType> acc(countTree);

...
acc.modifyValue(ijk, tools::valxform::SumOp<ValueType>(1));
...

```